lib/game/entities/unicorn/unicorn.dart

```diff
@@ -62,4 +70,79 @@ class Unicorn extends PositionedEntity with HasGameRef {
 
   /// Returns whether the animation is playing or not.
   bool isAnimationPlaying() => !animationTicker.done();
+
+  /// Moves the unicorn up by a specified amount.
+  void moveUp(double amount) {
+    position.y -= amount;
+    position.y = position.y.clamp(size.y / 2, gameRef.size.y - size.y / 2);
+  }
+
+  /// Moves the unicorn down by a specified amount.
+  void moveDown(double amount) {
+    position.y += amount;
+    position.y = position.y.clamp(size.y / 2, gameRef.size.y - size.y / 2);
+  }
```

Comment:
```suggestion
  void move(double amount) {
    position.y = amount;
    position.y = position.y.clamp(size.y / 2, gameRef.size.y - size.y / 2);
  }
```

Then when you want to move up, you just pass a negative value